lighthouse-core/gather/gatherers/dobetterweb/doctype.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('../gatherer');
+
+class Doctype extends Gatherer {
+  /**
+   * @param {LH.Gatherer.PassContext} passContext
+   * @return {Promise<LH.Artifacts['Doctype']>}
+   */
+  afterPass(passContext) {
+    const driver = passContext.driver;
+
+    return driver.getDocument().then(node => node);
```

Comment:
I am hoping this will return `document` as it does not look like I can call `driver.document` as what I am really after is `document` and then from that `document.name`

Reply:
rather than using the protocol's DOM methods, let's just grab the doctype via javascript.

define a function declaration in the gatherer.. something like 

```js
function getDoctype() {
  const {name, publicId, systemId} = document.doctype;
  return {name, publicId, systemId};
}
```

( i think we should grab the name, publicId, and systemId )

then, on L18 you can do 
```js
driver.evaluateAsync(`(${getDoctype.toString()}())`)
```

see the cache-contents gatherer for a model.